cumulusci/robotframework/Salesforce.py

```diff
@@ -0,0 +1,477 @@
+        res = obj_class.create(kwargs)
+        self.store_session_record(obj_name, res['id'])
+        return res['id']
+
+    def salesforce_query(self, obj_name, **kwargs):
+        """ Constructs and runs a simple SOQL query and returns the dict results """
+        query = 'SELECT '
+        if 'select' in kwargs:
+            query += kwargs['select']
+        else:
+            query += 'Id'
+        query += ' FROM {}'.format(obj_name)
+        where = []
+        for key, value in kwargs.items():
+            if key == 'select':
+                continue
+            where.append("{} = '{}'".format(key, value))
+        if where:
+            query += ' WHERE ' + ' AND '.join(where)
+        self.builtin.log('Running SOQL Query: {}'.format(query))
+        return self.cumulusci.sf.query_all(query).get('records', [])
+
+    def salesforce_update(self, obj_name, obj_id, **kwargs):
+        """ Updates a Salesforce object by id and returns the dict results """
+        self.builtin.log('Updating {} {} with values {}'.format(obj_name, obj_id, kwargs))
+        obj_class = getattr(self.cumulusci.sf, obj_name)
+        return obj_class.update(obj_id, kwargs)
+        
+    def soql_query(self, query):
+        """ Runs a simple SOQL query and returns the dict results """
+        self.builtin.log('Running SOQL Query: {}'.format(query))
+        return self.cumulusci.sf.query_all(query)
+
+    def store_session_record(self, obj_type, obj_id):
+        """ Stores a Salesforce record's id for use in the Delete Session Records keyword """
+        self.builtin.log('Storing {} {} to session records'.format(obj_type, obj_id))
+        self._session_records.append({
+            'type': obj_type,
+            'id': obj_id,
+        })
+
+    def wait_until_modal_is_open(self):
+        """ EXPERIMENTAL!!! """
+        self._call_selenium('_wait_until_modal_is_open', True)
+
+    def _wait_until_modal_is_open(self):
+        self.selenium.wait_until_element_is_not_visible(
+            "css: div.slds-spinner",
+            timeout=15,
+        )
+        self.selenium.wait_until_element_is_visible(
+            locators['modal']['is_open'],
+            timeout=15,
+        )
+        time.sleep(3)
+
+    def wait_until_modal_is_closed(self):
+        """ EXPERIMENTAL!!! """
+        self._call_selenium('_wait_until_modal_is_closed', True)
+
+    def _wait_until_modal_is_closed(self):
+        self.selenium.wait_until_element_is_not_visible(
+            "css: div.slds-spinner",
+            timeout=15,
+        )
+        self.selenium.wait_until_element_is_not_visible(
+            locators['modal']['is_open'],
+            timeout=15,
+        )
+        time.sleep(3)
+
+    def wait_until_loading_is_complete(self):
+        """ EXPERIMENTAL!!! """
+        self._call_selenium('_wait_until_loading_is_complete', True)
+
+    def _wait_until_loading_is_complete(self):
+        self.selenium.wait_until_element_is_not_visible(
```

Comment:
are these locators we can put in the locators.py file? it seems weird that they're here. idk much about robot.

Reply:
Don't have to know much about robot to notice that this doesn't make sense... I was just lazy :)